pkg/resourceinterpreter/customizedinterpreter/customized.go

```diff
@@ -32,8 +33,107 @@ type CustomizedInterpreter struct {
 	clientManager *webhookutil.ClientManager
 }
 
+// GetReplicas returns the desired replicas of the object as well as the requirements of each replica.
+// return matched value to indicate whether there is a matching hook.
+func (e *CustomizedInterpreter) GetReplicas(object *unstructured.Unstructured) (replica int32, requires *workv1alpha2.ReplicaRequirements, enabled bool, err error) {
+	klog.V(4).Infof("Get replicas for object: %v %s/%s with webhook interpreter.",
+		object.GroupVersionKind(), object.GetNamespace(), object.GetName())
+	var response *webhook.ResponseAttributes
+	response, enabled, err = e.interpret(context.TODO(), &webhook.RequestAttributes{
+		Operation: configv1alpha1.InterpreterOperationInterpretReplica,
+		Object:    object,
+	})
+	if err != nil {
+		return
+	}
+	if !enabled {
+		return
+	}
+	return response.Replicas, response.ReplicaRequirements, true, nil
+}
+
+// ReviseReplica revises the replica of the given object.
+func (e *CustomizedInterpreter) ReviseReplica(object *unstructured.Unstructured, replica int64) (obj *unstructured.Unstructured, enabled bool, err error) {
+	obj, enabled, err = e.Patch(context.TODO(), &webhook.RequestAttributes{
+		Operation:   configv1alpha1.InterpreterOperationReviseReplica,
+		Object:      object,
+		ReplicasSet: int32(replica),
+	})
+	return
+}
+
+// Retain returns the objects that based on the "desired" object but with values retained from the "observed" object.
+func (e *CustomizedInterpreter) Retain(desired *unstructured.Unstructured, observed *unstructured.Unstructured) (retained *unstructured.Unstructured, enabled bool, err error) {
+	retained, enabled, err = e.Patch(context.TODO(), &webhook.RequestAttributes{
```

Comment:
Seem we don't need to export the `Patch` method anymore, can you help update it?

Reply:
Done!